bundles/automation/org.eclipse.smarthome.automation.json.parsers/src/main/java/org/eclipse/smarthome/automation/parser/json/RuleJSONParser.java

```diff
@@ -0,0 +1,305 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.parser.json;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.automation.Action;
+import org.eclipse.smarthome.automation.AutomationFactory;
+import org.eclipse.smarthome.automation.Condition;
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.automation.handler.parser.Parser;
+import org.eclipse.smarthome.automation.handler.parser.Status;
+import org.eclipse.smarthome.automation.template.RuleTemplate;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class serves for loading JSON files and parse it to the Rule objects.
+ *
+ * @author Ana Dimova - Initial Contribution
+ *
+ */
+public class RuleJSONParser implements Parser {
+
+    private BundleContext bc;
+    private TemplateManager templateManager;
+    private ModuleTypeManager moduleManager;
+    private AutomationFactory automationFactory;
+    private Logger log;
+
+    public RuleJSONParser(BundleContext bc, TemplateManager templateManager, ModuleTypeManager moduleManager,
```

Comment:
I think we need some deeper discussion here.
In general my expectation was that a JSON parser won't have any dependency to the templateManager nor the moduleManager. It should simply read some JSON and create a Java object model with the help of the automationFactory.

Reading the code, I see only two parts where they are used:
1. For checking whether a module type or template already exists. I think we can remove this check as it should not be the parser who decides whether the parsed result gets added or not - it's not its business. (see also my comments at the Automation Provider code)
2. This is a bit more tricky: What I found is that the moduleTypeManager is used for serializing "outputs" of actions. Now on the Action API we only have connections defined and they only say what should be fed into the inputs. For a concrete action (not its type) there should imho no need to list outputs. If this is correct, there should not be a need to depend on the moduleTypeManager at all.

Am I missing anything?